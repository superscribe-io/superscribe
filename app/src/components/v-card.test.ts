import { test, expect } from 'vitest';
import { mount } from '@vue/test-utils';

import VCard from './v-card.vue';

test('Mount component', () => {
	expect(VCard).toBeTruthy();

	const wrapper = mount(VCard, {
		slots: {
			default: 'Slot Content',
		},
	});

	expect(wrapper.html()).toMatchSnapshot();
});

test('style props', () => {
	const props = ['disabled', 'tile'];

	for (const prop of props) {
		const wrapper = mount(VCard, {
			props: {
				[prop]: true,
			},
		});

		expect(wrapper.classes()).toContain(prop);
	}
});
