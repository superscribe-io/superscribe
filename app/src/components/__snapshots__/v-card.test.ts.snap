// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Mount component 1`] = `"<div data-v-12731031=\\"\\" class=\\"v-card\\">Slot Content</div>"`;
